tests/scale/virt-api.go

```diff
@@ -5,57 +5,41 @@ import (
 	"strconv"
 	"time"
 
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
 	v12 "kubevirt.io/api/core/v1"
 
 	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/flags"
 	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/util"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"kubevirt.io/client-go/kubecli"
 
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
 	"kubevirt.io/kubevirt/pkg/virt-operator/resource/generate/components"
-	"kubevirt.io/kubevirt/tests/util"
 )
 
 var _ = Describe("[sig-compute] virt-api scaling", decorators.SigCompute, func() {
 	var virtClient kubecli.KubevirtClient
 	numberOfNodes := 0
 
-	setccs := func(ccs v12.CustomizeComponents) (oldcss v12.CustomizeComponents) {
-		originalKv := util.GetCurrentKv(virtClient)
-		kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(context.Background(), originalKv.Name, v1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		oldcss = kv.Spec.CustomizeComponents
-		kv.Spec.CustomizeComponents = ccs
-		EventuallyWithOffset(1, func() error {
-			_, err = virtClient.KubeVirt(originalKv.Namespace).Update(context.Background(), kv, v1.UpdateOptions{})
+	setccs := func(ccs v12.CustomizeComponents, kvNamespace string, kvName string) error {
+		patchPayload, err := patch.New(patch.WithReplace("/spec/customizeComponents", ccs)).GeneratePayload()
+		if err != nil {
 			return err
-		}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
-
-		return oldcss
-	}
-
-	restorescc := func(ccs v12.CustomizeComponents) {
-		originalKv := util.GetCurrentKv(virtClient)
-		kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(context.Background(), originalKv.Name, v1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		kv.Spec.CustomizeComponents = v12.CustomizeComponents{}
-
-		EventuallyWithOffset(1, func() error {
-			_, err = virtClient.KubeVirt(originalKv.Namespace).Update(context.Background(), kv, v1.UpdateOptions{})
-			return err
-		}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+		_, err = virtClient.KubeVirt(kvNamespace).Patch(context.Background(), kvName, types.JSONPatchType, patchPayload, v1.PatchOptions{})
+		return err
 	}
 
 	getApiReplicas := func(virtClient kubecli.KubevirtClient, expectedResult int32) int32 {
 		By("Finding out virt-api replica number")
-		kv := util.GetCurrentKv(virtClient)
-		Expect(kv).ToNot(BeNil())
-
-		apiDeployment, err := virtClient.AppsV1().Deployments(kv.GetNamespace()).Get(context.Background(), components.VirtAPIName, v1.GetOptions{})
+		apiDeployment, err := virtClient.AppsV1().Deployments(flags.KubeVirtInstallNamespace).Get(context.Background(), components.VirtAPIName, v1.GetOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
 		By("Expecting number of replicas to be as expected")
```

Comment:
"Expecting number of replicas to not be nil"

Reply:
Thanks, fixed